### PR TITLE
Windows support

### LINKED ISSUE #114: Windows support
#### Windows

The Windows version needs someone to implement process handling!
Right now there's pretty much no Windows specific code which means:

- [ ] Pausing/resuming commands doesn't work for now.
- [ ] Sending signals to processes' children doesn't work.
- [ ] We cannot ensure that there aren't dangling processes on `kill`.

### PR DESCRIPTION
**Pull request by oiatz**
- Pausing/resuming commands works for now.
- Always sending signals to processes' children.
- Get all processes before `kill` command, so we can ensure that there aren't dangling processes on `kill`

Resolves #114

### CODE REVIEW

**Review by Nukesor — COMMENTED** [review]
Hi!
Thanks for taking the time to tackle this issue :+1: 

I'll try to review this in the next few days. It'll probably take a little while, though.
This whole issue requires some research and reading on my end :)

**Review by Nukesor — CHANGES_REQUESTED** [review]
Thanks again for tackling this issue.
Most comments regard documentation, but there are also a few open questions on how Window's process/thread signaling works.

It would also be awesome, if you could update the Readme:
```
Works on Linux and **partially** on MacOS and Windows. \
```
to something like:
```
- Linux is fully supported and battle-tested.
- Windows has all features, but process handling is **experimental**.
- MacOS only has **rudimentary** process handling (A few things don't work).
    It's still usable, though.
```

Also, don't forget to add your change to the `CHANGELOG.md`, reference your Github profile and link the issue (https://github.com/Nukesor/pueue/issues/114).
I just added the next version to the `CHANGELOG.md` on master. Just rebase/merge and you should be good to go :)

File: daemon/platform/windows/process_helper.rs [hunk 1]
```diff
@@ -17,15 +28,39 @@ pub fn compile_shell_command(command_string: &str) -> Command {
 
 /// Send a signal to a windows process.
 pub fn send_signal_to_child(
-    _child: &Child,
+    child: &Child,
     action: &ProcessAction,
     _children: bool,
```
[the remark this answers is not in the document]
Comment (oiatz) [reply daemon/platform/windows/process_helper.rs:33]: AFAIK, Windows doesn't provide SIGSTOP or SIGCONT signals [link](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/signal?view=msvc-160)。

So we have to enumerate the set of processes and threads, then call `SuspendThread` or `ResumeThread` to achieve the same effect. But this operation may carry some risk. For example, while enumerating the set of threads, new threads can be created and destroyed. I thought it's at user's risk.

**Review by Nukesor — APPROVED** [review]

**Comment by Nukesor** [comment]
Changes look good to me :)!

There's a tiny conflict in the changelog. As soon as this is resolved, I'll go ahead and merge the PR.
Thanks again for implementing this,

Cheers!

**Pull request merged**